### PR TITLE
Dashboards admin

### LINKED ISSUE #3455: Admin / Dashboards permissions
Add new actions in Groups and permissions, under the Plans module: 

- "Manage dashboards" - Users with that permission can: 
  - Manage dashboards (CRUD, edit content of a dashboard)
  - See the Dashboards tab in a Plan
- "View Dashboards" - Users with that permission can:
  - See the Dashboards tab in a Plan

### LINKED ISSUE #3456: Admin / Dashboards list
- New tab in the admin page of a Plan: "Dashboards", after Projects
- In Plan/Dashboards:
  - Page title: "Dashboards"
  - Action button (in the left): New
  - Table with (columns): 
    - Edit icon
    - Dashboard [title, linked to Dashboard page]
    - Visibility [Private, Team, Public]
    - ICON View item
    - ICON Public view (only if Dashboard configured as Public)

### PR DESCRIPTION
**Pull request by entantoencuanto**
Closes #3455
Closes #3456

## :v: What does this PR do?

* Adds new permissions to admin permissions groups to manage and view dashboards.
* Adds a new Dashboards tab to plans (only available if the admin has the right permissions)
* Displays an index of dashboards of a plan and a button to create new dashboards (maybe this action should be managed with the front app). For the moment the only attribute available is the title of the dataset, the widgets configuration is expected to be managed via API
* Adds links in dashboards of index to open in a modal window the viewer and maker divs to start the front application with the correct context and id. For some reason the modal opens but nothing happens.

**NOTE** This PR is built on top of #3584.

## :shipit: Does this PR changes any configuration file?

No

(Changes in these files might need to update the role in Ansible)

## :book: Does this PR require updating the documentation?

No

### CODE REVIEW

**Comment by Crashillo** [comment]
https://github.com/PopulateTools/gobierto/blob/cd2a2a734429b4f3c1da51e8f56393c7a1a1ea1d/app/views/gobierto_plans/plan_types/show.html.erb#L36-L43

**IMPORTANT**: replace `data-context` and `data-pipe` to be set from the rails controller. In that way, it will be transparent for javascript.

File: app/javascript/gobierto_dashboards/webapp/Maker.vue [hunk 1]
```diff
@@ -26,6 +25,7 @@
           <input
             id="dashboard-status"
             type="checkbox"
+            :checked="status"
```
[the remark this answers is not in the document]
Comment (Crashillo) [reply app/javascript/gobierto_dashboards/webapp/Maker.vue:28]: This is grating me. In local dev, running `eslint app/javascript/gobierto_dashboards/webapp/Maker.vue` does not display any kind of error.

According its docs, Hound uses v4.18.2 http://help.houndci.com/en/articles/2138530-eslint, what it sounds quite odd that `eslint-plugin-vue` works, as its website specifies they require eslint > 6.x https://eslint.vuejs.org/user-guide/#installation (current eslint version `7.x` ....)

🤯

[the remark this answers is not in the document]
Comment (Crashillo) [reply app/javascript/gobierto_dashboards/webapp/Maker.vue:28]: Running same local eslint version (v.4.18.2) produces nothing. 🤯 💥

**Comment by Crashillo** [comment]
Let's keep this branch as it is, new features/bugs will be built on top of this

**Comment by entantoencuanto** [comment]
> https://github.com/PopulateTools/gobierto/blob/cd2a2a734429b4f3c1da51e8f56393c7a1a1ea1d/app/views/gobierto_plans/plan_types/show.html.erb#L36-L43
> 
> **IMPORTANT**: replace `data-context` and `data-pipe` to be set from the rails controller. In that way, it will be transparent for javascript.

Done in https://github.com/PopulateTools/gobierto/pull/3623

**Comment by entantoencuanto** [comment]
Test coverage drops but it has to do with controller actions that are going to be removed

**Review by ferblape — APPROVED** [review]

**Pull request merged**